src/api/Feed.js

```diff
@@ -615,18 +622,63 @@ class Feed extends Base {
-            task: taskPayload,
-            successCallback: (taskData: Task) => {
-                this.addPendingItem(this.file.id, currentUser, pendingTask);
-                this.createTaskNewSuccessCallback(file, uuid, taskData, assignees, successCallback, errorCallback);
-            },
-            errorCallback: (e: ElementsXhrError, code: string) => {
-                this.feedErrorCallback(false, e, code);
-            },
+        // Pull out groups from assignees
+        const groupAssignees = assignees.filter(
+            (assignee: SelectorItem<UserMini | GroupMini>) => (assignee.item && assignee.item.type) === 'group',
+        );
+
+        // Pull out group id from groupAssignees array
+        const groupIds = groupAssignees.map(assignee => assignee.id);
+
+        // run API call against each id --> returns an array of API calls (promises)
+
+        const groupInfoPromises: Array<Promise<any>> = groupIds.map(groupId => {
+            const groupsAPI = new GroupsAPI(this.options);
+            return groupsAPI.getGroupCount({
+                file,
+                group: { id: groupId },
+            });
         });
+
+        // Fetch each group size in parallel --> return an array of group sizes
+        Promise.all(groupInfoPromises)
+            .then((groupCounts: Array<{ total_count: number }>) => {
+                const hasAnyGroupCountExceeded: boolean = groupCounts.some(
+                    groupInfo => groupInfo.total_count > GroupsAPI.MAX_GROUP_ASSIGNEES,
+                );
+                const warning = {
+                    code: ERROR_CODE_GROUP_EXCEEDS_LIMIT,
+                    type: 'warning',
+                };
+
+                // If any group count exceeds 250, throw an error
+                if (hasAnyGroupCountExceeded) {
+                    this.feedErrorCallback(false, warning, ERROR_CODE_GROUP_EXCEEDS_LIMIT);
+                } else {
+                    this.tasksNewAPI = new TasksNewAPI(this.options);
+                    this.tasksNewAPI.createTask({
+                        file,
+                        task: taskPayload,
+                        successCallback: (taskData: Task) => {
+                            this.addPendingItem(this.file.id, currentUser, pendingTask);
+                            this.createTaskNewSuccessCallback(
+                                file,
+                                uuid,
+                                taskData,
+                                assignees,
+                                successCallback,
+                                errorCallback,
+                            );
+                        },
+                        errorCallback: (e: ElementsXhrError, code: string) => {
+                            this.feedErrorCallback(false, e, code);
+                        },
+                    });
+                }
+            })
+            .catch(error => {
+                console.error('task has not been created', error);
```

Comment:
What kind of error do we want to throw in the `catch`? @booc0mtaco @alexkrolick

Reply:
here, let's mimick what we do with the `errorCallback` instances

want to use `this.feedErrorCallback`